src/ui/scanpopup.cc

```diff
@@ -58,9 +58,44 @@ ScanPopup::ScanPopup( QWidget * parent,
   dictionaryBar( this, configEvents, cfg.preferences.maxDictionaryRefsInContextMenu ),
   hideTimer( this )
 {
-  ui.setupUi( this );
-  toolbar = new QToolBar( "Found Dictionary", this );
-  toolbar->setObjectName( "foundDictionary" );
+  // Init UI
+  QWidget * toolBarWidget = new QWidget( this );
+  ui.setupUi( toolBarWidget );
+
+  QToolBar * toolBar = new QToolBar( "Tool bar", this );
+  toolBar->setObjectName( "popupToolBar" );
+  toolBar->addWidget( toolBarWidget );
+
+  groupList    = new GroupComboBox( this );
+  translateBox = new TranslateBox( this );
+
+  QToolBar * searchBar = new QToolBar( "Search bar", this );
+  searchBar->setObjectName( "popupSearchBar" );
+  groupListAction = searchBar->addWidget( groupList );
+  searchBar->addWidget( translateBox );
+  searchBar->toggleViewAction()->setEnabled( false );
+
+  foundBar = new QToolBar( "Navgiation bar", this );
+  foundBar->setObjectName( "popupNavgiationBar" );
+  // to match the articleView's vertial scrolling
+  foundBar->setAllowedAreas( Qt::LeftToolBarArea | Qt::RightToolBarArea );
+
+  // UI style
+  searchBar->setSizePolicy( QSizePolicy::MinimumExpanding, QSizePolicy::Maximum );
+  searchBar->setMovable( false );
+  toolBar->setFloatable( false );
+  dictionaryBar.setFloatable( false );
+  foundBar->setFloatable( false );
+
+  searchBar->setContentsMargins( 0, 0, 0, 0 );
```

Comment:
Maybe we can add 1 more pixel here.

It is only a problem of the vista theme (current default on Windows) due to the border is thin and the them is very rectangular.